controllers/falcon_container/falconcontainer_controller.go

```diff
@@ -0,0 +1,195 @@
+			// Return and don't requeue
+			return ctrl.Result{}, nil
+		}
+		// Error reading the objectclusterRole := r.newClusterRole() - requeue the request.
+		r.Log.Error(err, "cannot get the Falcon Container custom resource")
+		return ctrl.Result{}, err
+	}
+	if falconContainer.Status.Phase != v1alpha1.PhaseDone {
+		r.UpdateStatus(ctx, req, falconContainer, v1alpha1.PhaseReconciling)
+	}
+
+	r.Log.Info("Reconciling Namespace")
+	if _, err := r.reconcileNamespace(ctx, falconContainer); err != nil {
+		r.Error(ctx, req, falconContainer, fmt.Sprintf("failed to reconcile namespace: %v", err))
+		return ctrl.Result{}, fmt.Errorf("failed to reconcile namespace: %v", err)
+	}
+
+	switch falconContainer.Spec.Registry.Type {
+	case v1alpha1.RegistryTypeECR:
+		r.Log.Info("Reconciling ECR Repository")
+		if _, err := r.UpsertECRRepo(ctx); err != nil {
+			r.Error(ctx, req, falconContainer, fmt.Sprintf("failed to reconcile ECR repository: %v", err))
+			return ctrl.Result{}, fmt.Errorf("failed to reconcile ECR repository: %v", err)
+		}
+	case v1alpha1.RegistryTypeOpenshift:
+		r.Log.Info("Reconciling Image Stream")
+		stream, err := r.reconcileImageStream(ctx, falconContainer)
+		if err != nil {
+			r.Error(ctx, req, falconContainer, fmt.Sprintf("failed to reconcile Image Stream: %v", err))
+			return ctrl.Result{}, fmt.Errorf("failed to reconcile Image Stream")
+		}
+		if stream == nil {
+			return ctrl.Result{}, nil
+		}
+	}
+	if r.imageMirroringEnabled(falconContainer) {
+		r.Log.Info("Verifying image availability in remote registry")
+		if err := r.PushImage(ctx, falconContainer); err != nil {
+			r.Error(ctx, req, falconContainer, fmt.Sprintf("failed to refresh Falcon Container image: %v", err))
+			return ctrl.Result{}, fmt.Errorf("cannot refresh Falcon Container image: %v", err)
+		}
+	} else {
+		r.Log.Info("Verifying access to CrowdStrike Container Image Registry")
+		updated, err := r.verifyCrowdStrikeRegistry(ctx, falconContainer)
+		if updated {
+			return ctrl.Result{}, nil
+		}
+		if err != nil {
+			r.Error(ctx, req, falconContainer, fmt.Sprintf("failed to verify CrowdStrike Container Image Registry access: %v", err))
+			return ctrl.Result{}, fmt.Errorf("failed to verify CrowdStrike Container Image Registry access")
+		}
+		r.Log.Info("Reconciling Container Registry pull token Secrets")
+		if _, err = r.reconcileRegistrySecrets(ctx, falconContainer); err != nil {
+			r.Error(ctx, req, falconContainer, fmt.Sprintf("failed to reconcile Falcon registry pull token Secrets: %v", err))
+			return ctrl.Result{}, fmt.Errorf("failed to reconcile Falcon registry pull token Secrets: %v", err)
+		}
+	}
+
+	r.Log.Info("Reconciling ServiceAccount")
+	if _, err := r.reconcileServiceAccount(ctx, falconContainer); err != nil {
+		r.Error(ctx, req, falconContainer, fmt.Sprintf("failed to reconcile Service Account: %v", err))
+		return ctrl.Result{}, fmt.Errorf("failed to reconcile Service Account: %v", err)
+	}
```

Comment:
This needs to be in `config/rbac/container_service_account.yaml`

Reply:
Will be addressed in a future PR.